### PR TITLE
Discourage use of conda and recommend virtualenv

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Discourage use of conda and recommend virtualenv.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [x] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -1646,29 +1646,28 @@ MLflow currently supports the following environment management tools to restore
...
     Create environments using virtualenv and pyenv (for python version management). Virtualenv and
     pyenv (for Linux and macOS) or pyenv-win (for Windows) must be installed for this mode of environment reconstruction.
 
     - `virtualenv installation instructions <https://virtualenv.pypa.io/en/latest/installation.html>`_
     - `pyenv installation instructions <https://github.com/pyenv/pyenv#installation>`_
     - `pyenv-win installation instructions <https://github.com/pyenv-win/pyenv-win#installation>`_
 
-    .. note::
-        Virtualenv support is still experimental and may be changed in a future MLflow release.
+conda (discouraged)
```
Comment (dbczumar) [inline docs/source/models.rst:1658]:
```suggestion
conda
```

I think dropping `discouraged` is fine as long as we add a warning

File: docs/source/models.rst [hunk 2]
```diff
@@ -1646,29 +1646,28 @@ MLflow currently supports the following environment management tools to restore
...
+    Create environments using conda. Conda must be installed for this mode of environment reconstruction.
```
Comment (dbczumar) [inline docs/source/models.rst:1659]: Can we also tell users that, by using conda, the user is responsible for adhering to Anaconda's terms of use?

File: docs/source/projects.rst [hunk 3]
```diff
@@ -71,38 +71,11 @@ that is used to execute project entry points.
...
-
-  .. code-block:: bash
-
-      mlflow run /path/to/conda/project --env-manager virtualenv
-
-  .. warning::
-
-      When a conda environment project is executed as a virtualenv environment project,
-      conda dependencies will be ignored and only pip dependencies will be installed.
+MLflow currently supports the following project environments: Virtualenv environment, Conda environment, Docker container environment, and system environment.
```
Comment (dbczumar) [inline docs/source/projects.rst:74]: Nittiest of nits:

```suggestion
MLflow currently supports the following project environments: Virtualenv environment, conda environment, Docker container environment, and system environment.
```

File: docs/source/projects.rst [hunk 4]
```diff
@@ -139,6 +112,33 @@ Docker container environment
   a Docker container environment in an ``MLproject`` file, see
   :ref:`mlproject-specify-environment`.
 
+.. _project-conda-environments:
+
+Conda environment (discouraged)
```
Comment (dbczumar) [inline docs/source/projects.rst:117]:
```suggestion
Conda environment
```

Below, can we also tell users that, by using conda, the user is responsible for adhering to Anaconda's terms of use?

File: mlflow/utils/cli_args.py [hunk 5]
```diff
@@ -60,11 +61,15 @@
 def _resolve_env_manager(_, __, env_manager):
     if env_manager is not None:
         _EnvManager.validate(env_manager)
-        if env_manager == _EnvManager.VIRTUALENV:
+        if env_manager == _EnvManager.CONDA and not MLFLOW_DISABLE_ENV_MANAGER_CONDA_WARNING.get():
             warnings.warn(
                 (
-                    "Virtualenv support is still experimental and may be changed in a future "
-                    "release without warning."
+                    "Use of Anaconda is discouraged. If you use it, please ensure that your use of "
```
Comment (dbczumar) [inline mlflow/utils/cli_args.py:67]:
```suggestion
                    "Use of conda is discouraged. If you use it, please ensure that your use of "
```

File: mlflow/utils/cli_args.py [hunk 6]
```diff
@@ -60,11 +61,15 @@
...
+                    "Anaconda complies with their terms of service "
+                    "(https://legal.anaconda.com/policies/en/?name=terms-of-service). "
+                    "virtualenv is recommended as an alternative. "
```
Comment (dbczumar) [inline mlflow/utils/cli_args.py:70]:
```suggestion
                    "virtualenv is the recommended tool for environment reproducibility. "
```

**Review by dbczumar — APPROVED** [review]
@harupy Awesome stuff! LGTM with a few comments!

**Review by WeichenXu123 — APPROVED** [review]
LGTM once @dbczumar 's comments being addressed.

File: docs/source/models.rst (hunk 1 above)
Comment (harupy) [reply docs/source/models.rst:1658]: Got it, removed!

File: docs/source/projects.rst (hunk 4 above)
Comment (harupy) [reply docs/source/projects.rst]: Removed `(discouraged)`!

File: docs/source/projects.rst (hunk 3 above)
Comment (harupy) [reply docs/source/projects.rst]: Fixed!

File: mlflow/utils/cli_args.py (hunk 6 above)
Comment (harupy) [reply mlflow/utils/cli_args.py]: Fixed!

File: mlflow/utils/cli_args.py (hunk 5 above)
Comment (harupy) [reply mlflow/utils/cli_args.py]: Fixed!

File: docs/source/models.rst [hunk 7]
```diff
@@ -1646,29 +1646,32 @@ MLflow currently supports the following environment management tools to restore
...
     - `pyenv-win installation instructions <https://github.com/pyenv-win/pyenv-win#installation>`_
 
-    .. note::
-        Virtualenv support is still experimental and may be changed in a future MLflow release.
+conda
+    Create environments using conda. Conda must be installed for this mode of environment reconstruction.
+
+    .. warning::
+
+        By using conda, you're responsible for adhering to `Anaconda's terms of service <https://legal.anaconda.com/policies/en/?name=terms-of-service>`_.
```
Comment (harupy) [inline docs/source/models.rst:1663]: @dbczumar Added this warning to address [the comment above](https://github.com/mlflow/mlflow/pull/6507/files#r950651512).

File: docs/source/models.rst (hunk 2 above)
Comment (harupy) [reply docs/source/models.rst:1659]: Added!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**